api-lib/gql/userClient.ts

```diff
@@ -0,0 +1,15 @@
+import { NODE_HASURA_URL } from '../config';
```

Comment:
This file can probably be deleted, right

Reply:
I believe not, this file is to run a query impersonating as a user, looking at the test `api-test/__tests__/deleteContribution.test.ts`

```typescript
const client = userClient({
  profileId: profile.id,
  address: ADDRESS,
});

const contribution = await createContribution(client, {
  circle_id: circle.id,
  user_id: user.id,
});
```
I'm creating a GQL client impersonating as a user in order to create a contribution as them, then in the `api-lib/gql/makeThunder.ts` we map that to the headers that go to hasura as if we are the user. Without this file, we can only making requests as an admin, which will not allow us to test for example permissions as a normal user. Or create multiple clients impersonating different users interacting with one another.
```typescript
const mapOptionsToHeaders = (options: UserOptions) => ({
  'x-hasura-role': 'user',
  'x-hasura-address': options.address,
  'x-hasura-user-id': String(options.profileId),
});

export const makeThunder = ({ url, options }: ThunderOptions) => {
  return Thunder(async (...params) => {
    return apiFetch([
      url,
      {
        method: 'POST',
        headers:
          'adminSecret' in options
            ? {
                'x-hasura-admin-secret': options.adminSecret,
              }
            : {
                ...mapOptionsToHeaders(options),
                authorization: 'test-skip-auth',
              },
      },
    ])(...params);
  });
};

```

Reply:
@karelianpie thanks for that explanation! Since it isn't supposed to be used in production, can we rename it `mockUserClient`?

Reply:
Good idea @topocount, renamed with https://github.com/coordinape/coordinape/pull/1486/commits/69dcf95aee4424365953c6a71eca723f762e1409